compiler/src/iree/compiler/Codegen/Passes.td

```diff
@@ -47,6 +47,28 @@ def DecomposeLinalgGeneric :
   let constructor = "mlir::iree_compiler::createDecomposeLinalgGenericPass()";
 }
 
+def DecomposeAffineOps: Pass<"decompose-affine-ops"> {
+  let summary = "TODO";
+  let description = [{
+     TODO
+  }];
+  let constructor = "mlir::iree_compiler::createDecomposeAffineOpsPass()";
+  let dependentDialects = [
+      "AffineDialect"
+  ];
+}
+
+def ExtractAddressComputation: Pass<"extract-address-computation"> {
+  let summary = "TODO";
+  let description = [{
+     TODO
+  }];
+  let constructor = "mlir::iree_compiler::createExtractAddressComputationPass()";
+  let dependentDialects = [
+      "memref::MemRefDialect", "nvgpu::NVGPUDialect"
```

Comment:
Could we decouple the dependence on `NVGPUDialect`. So a pass that can run on CPU without that dependency, and a pass with potentially extra patterns that are needed for `NVGPU`

Reply:
Sure thing

Reply:
I've split the patterns into two passes now.
The GPU pass is a super set of the "generic" pass for now.

Let me know if that works or if you prefer to have the generic and GPU pass being disjointed.

Also let me know if you want them to live in different files.

Finally, I didn't know if it was okay to leave the GPU one in `Common`, but saw that some GPU passes were already there (e.g., `GPUPipelining.cpp`).